providers/dns/stackpath/stackpath.go

```diff
@@ -0,0 +1,121 @@
+package stackpath
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/xenolf/lego/acme"
+	"github.com/xenolf/lego/platform/config/env"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	defaultBaseURL = "https://gateway.stackpath.com/dns/v1/stacks"
+	defaultAuthURL = "https://gateway.stackpath.com/identity/v1/oauth2/token"
+)
+
+// DNSProvider is an implementation of the acme.ChallengeProvider interface.
+type DNSProvider struct {
+	client *http.Client
+	config *Config
+}
+
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	StackID            string
+	TTL                int
+	PropagationTimeout time.Duration
+	PollingInterval    time.Duration
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	return &Config{
+		TTL:                env.GetOrDefaultInt("STACKPATH_TTL", 120),
+		PropagationTimeout: env.GetOrDefaultSecond("STACKPATH_PROPAGATION_TIMEOUT", acme.DefaultPropagationTimeout),
+		PollingInterval:    env.GetOrDefaultSecond("STACKPATH_POLLING_INTERVAL", acme.DefaultPollingInterval),
+	}
+}
+
+// NewDNSProviderCredentials uses the supplied credentials
+// to return a DNSProvider instance configured for Stackpath.
+func NewDNSProviderCredentials(clientID, clientSecret, stackID string) (*DNSProvider, error) {
```

Comment:
You must remove this function and replace with `func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {}`

I recommend to create a client struct:
```go
type Client struct {
	clientID     string
	clientSecret string
	stackID      string
	BaseURL      string
}
```